crates/ty_python_semantic/src/types/property_tests.rs

```diff
@@ -313,16 +311,4 @@ mod flaky {
         bottom_materialization_of_type_is_assigneble_to_type, db,
         forall types t. t.bottom_materialization(db).is_assignable_to(db, t)
     );
-
-    // Any type assignable to `Iterable[object]` should be considered iterable.
-    //
-    // Note that the inverse is not true, due to the fact that we recognize the old-style
-    // iteration protocol as well as the new-style iteration protocol: not all objects that
-    // we consider iterable are assignable to `Iterable[object]`.
-    //
-    // Currently flaky due to <https://github.com/astral-sh/ty/issues/889>
-    type_property_test!(
-        all_type_assignable_to_iterable_are_iterable, db,
-        forall types t. t.is_assignable_to(db, KnownClass::Iterable.to_specialized_instance(db, [Type::object(db)])) => t.try_iterate(db).is_ok()
-    );
```

Comment:
I'm not convinced by the rationale given in the PR description for removing this property test. I think there are many type properties that we want to verify for valid types, that can be invalidated by an invalid type (e.g. a type that violates Liskov), and we probably should never introduce such types into our type generation for the property tests. Or if we do, we should still keep some property tests with a pre-condition that they only apply to valid types.

Especially if there's no _current_ problem, I would not remove this property test. If anything, I'd just add a note to its docstring that it is not true for types which explicitly inherit `Iterable` but violate LSP on its interface.